code-studio/src/js/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -165,20 +168,22 @@ var WorkshopForm = React.createClass({
   },
 
   handleSessionsChange: function (sessions, removedSession) {
-    this.state.sessions = sessions;
-    this.state.sessionsModified = true;
+    sessions.sessionsModified = true;
```

Comment:
This *might* actually get you into trouble still. It looks like `handleSessionsChange` is passed to `SessionListFormPart` as `onChange`. `SessionListFormPart` then calls it's change handler with `this.props`, which means when we mutate sessions.sessionsModified here, we're also mutating the props object of our SessionListFormPart. It's possible that React will even throw an error or warning about this.

Easiest fix might be to be the same thing we did elsewhere and clone sessions before mutating it. Longer term approach might be to start using something like immutable.js